\section{Introduction}
 
MetaOmics software suite is an interactive software with graphical user interface (GUI) for genomic meta-analysis implemented using R Shiny.
Many state of art meta-analysis tools are available in this software,
including MetaProcess for omics data preprocessing, 
MetaQC for quality control, 
MetaDE for differential expression analysis,
MetaPath for pathway enrichment analysis,
MetaNetwork for differential co-expression network analysis,
MetaPredict for classification analysis,
MetaClust for sparse clustering analysis,
MetaPCA for principal component analysis.

In this tutorial, 
we will go through installation and usage of MetaOmics step by step using real data examples.
The MetaOmics suite software is publicly available at \url{https://github.com/metaOmics/metaOmics}.
The tutorial itself can be found at \url{https://github.com/metaOmics/tutorial/blob/master/metaOmics_turtorial.pdf}.
Each MetaOmics module will be introduced in later sections and their R packages are also available on GitHub \url{https://github.com/metaOmics}.


\subsection{Abbreviation terms}

\begin{itemize}
\item General terms:
\begin{itemize}
\item CV: Cross validation
\item DE:  Differentially expressed
\item FDR: False discovery rate
\item FC: Fold change
\item FPKM: Fragments Per Kilobase Million mapped reads
\item QC: Quality control
\item RPKM: Read Per Kilobase Million mapped reads
\item TPM: Transcripts Per Kilobase Million mapped reads
\end{itemize}

\item Methods or tools:
\begin{itemize}
\item AW-Fisher: Adaptively weighted Fisher's method
\item CPI: Comparative pathway integrator 
\item FEM: Fixed effects model
\item K-S test: Kolmogorov-Smirnov test
\item MAPE: Meta analysis pathway enrichment method
\item PCA: Principal component analysis
\item REM: Random effects model
\item SMR: Standardized mean ranks
\item TSP: Top scoring pair algorithm

\end{itemize}

\end{itemize}



