\section{Introduction}
 
MetaOmics is a browser-based software suite for transcriptomic meta-analysis with R shiny-based graphical user interface (GUI).
Many state-of-the-art meta-analysis tools are available in this software,
including %MetaProcess for omics data preprocessing, 
MetaQC for quality control, 
MetaDE for differential expression analysis,
MetaPath for pathway enrichment analysis,
MetaNetwork for differential co-expression network analysis,
MetaPredict for classification analysis,
MetaClust for sparse clustering analysis and 
MetaPCA for principal component analysis.

In this tutorial, 
we will go through the installation and usage of MetaOmics step by step using real data examples.
The MetaOmics software suite is publicly available at \url{https://github.com/metaOmics/metaOmics}.
This tutorial can be found at \url{https://github.com/metaOmics/tutorial/blob/master/metaOmics_turtorial.pdf}.
Each MetaOmics module will be introduced in later sections and the associated R packages are available on GitHub at \url{https://github.com/metaOmics}.


\subsection{Abbreviation terms}



\begin{itemize}
\item General terms:
\begin{itemize}
\item CV: Cross validation
\item DE:  Differentially expressed
\item FDR: False discovery rate
\item FC: Fold change
\item FPKM: Fragments Per Kilobase Million mapped reads
\item QC: Quality control
\item RPKM: Read Per Kilobase Million mapped reads
\item TPM: Transcripts Per Kilobase Million mapped reads
\end{itemize}

\item Methods or tools:
\begin{itemize}
\item AW-Fisher: Adaptively weighted Fisher's method
\item CPI: Comparative pathway integrator 
\item FEM: Fixed effects model
\item K-S test: Kolmogorov-Smirnov test
\item MAPE: Meta analysis pathway enrichment method
\item PCA: Principal component analysis
\item REM: Random effects model
\item SMR: Standardized mean ranks
\item TSP: Top scoring pair algorithm

\end{itemize}

\end{itemize}

\subsection{Statistical method explanations}

{\color{red}
\begin{itemize}
\item
Think about including a table to briefly introduce (like what we did in the Supplementary information) the statistical terms such as KNN algorithm, 
Fisher?s exact test, KS test, FDR, consensus clustering, simulated annealing algorithm, etc. (ask George if necessary)
\item
Ask the users to refer to our MetaOmics paper (Supplementary information Box S2) for detailed explanation/interpretation of statistical learning approaches/terms, e.g. DE analysis, pathway analysis, differential co-expression network analysis, etc.
\end{itemize}

}

